src/components/TaskDetails/TaskDetails.jsx

```diff
@@ -46,94 +40,98 @@ export default class TaskDetails extends React.Component {
-      <Container className="task-details pl-0">
-        {'Name:'}
-        <Button
-          className="close-task-button cancel-button float-right px-0"
-          onClick={closeTaskDetails}
-        >
-          {'╳'}
-        </Button>
-        <Container className="task-details-container main-modal-for-taskN">
-          {nameContext}
-        </Container>
-        {'Description:'}
-        <Container className="task-details-container description-container main-modal-for-taskD">
-          {descriptionContext}
-        </Container>
-        <ButtonGroup>
-          <TaskStatus
-            status={status}
-            changeStatus={this.handleSaveTaskDetails}
+      const nameContext = editName
+        ? (
+          <EditName
+            name={taskDetails.name}
+            editName={editName}
+            closeEditNameField={this.handleEditName}
+            saveName={this.handleSaveTaskDetails}
           />
-          <TaskPriority
-            priority={priority}
-            changePriority={this.handleSaveTaskDetails}
+        ) : (
+          <Container
+            className="open-edit-name"
+            onClick={this.handleEditName}
+          >
+            {taskDetails.name}
+          </Container>
+        );
+      const descriptionContext = editDescription
+        ? (
+          <EditDescription
+            description={taskDetails.description}
+            editDescription={editDescription}
+            closeEditDescriptionField={this.handleEditDescription}
+            saveDescription={this.handleSaveTaskDetails}
           />
-        </ButtonGroup>
-      </Container>
-    );
-  }
+        ) : (
+          <Container
+            className="open-edit-description"
+            onClick={this.handleEditDescription}
+          >
+            {taskDetails.description || 'Some text'}
+          </Container>
+        );
+
+      return (
+        <Container className="task-details pl-0">
+          {'Name:'}
+          <Button
+            className="close-task-button cancel-button float-right px-0"
+            onClick={closeTaskDetails}
+          >
+            <img src={del} alt="close" className="delete-icon" />
+          </Button>
+          <Container className="task-details-container main-modal-for-taskN">
+            {nameContext}
+          </Container>
+          {'Description:'}
+          <Container className="task-details-container description-container main-modal-for-taskD">
+            {descriptionContext}
+          </Container>
+          <ButtonGroup>
+            <TaskStatus
+              status={taskDetails.status}
+              changeStatus={this.handleSaveTaskDetails}
+            />
+            <TaskPriority
+              priority={taskDetails.priority}
+              changePriority={this.handleSaveTaskDetails}
+            />
+          </ButtonGroup>
+        </Container>
+      );
+    }
 }
+
+const mapStateToProps = state => ({
+  taskDetails: state.taskDetailsReducer.taskDetails,
+});
```

Comment:
You can also write it like this:
`const mapStateToProps = ({ taskDetailsReducer: { taskDetails } }) => ({ taskDetails });`